### DISCUSSION TITLE
Subscriptions and observables in AppRouter Server components

### DISCUSSION THREAD

**Discussion by Boimb** [vercel/next.js#51035] [App Router]
Hi,
New to Nextjs I try to use the latest v13 and embrace the Server/Client Components paradigm, basically, fetching my data in a Server component.
I'm currently facing 2 issues:
- As there seems to be no "life cycle" methods in the server side, I can't manage to unsubscribe to the service I'm getting events from.
- Using an observable to make this data availbale to Client components I can't manage to keep this (Observable) on the server side, because observables are not serialiazable.
Consequence is that I'm basically keeping all this logic on the client side, thing I'd like to avoid.

I know all of this is quite new, but, are there any tricks out there to handle this? 

Thx for your help

**Comment by phaux** [comment]
It would be nice if Observable or AsyncIterable could be passed to client components via prop and then used via `use` hook. If your DB library already returns one of these then you get a real time app for free:


```tsx
const kv = await Deno.openKv("my_db.sqlite")

export default function Page() {
  const updates = kv.watch(["rooms", 123]) // returns AsyncIterable
 
  return (
    <Suspense fallback={<Spinner />}>
      <RoomView updates={updates} />
    </Suspense>
  )
}
```

```tsx
'use client'
 
export function RoomView(props: { updates: AsyncIterable<RoomData> }) {
  const room = use(props.updates)

  return <>
    <h1>Welcome to room {room.title}!</h1>
    <p>We have {room.members.length} participants.</p>
  </>
}
```
